pkg/util/validation/limits.go

```diff
@@ -885,3 +895,100 @@ func EnabledByAnyTenant(tenantIDs []string, f func(string) bool) bool {
 	}
 	return false
 }
+
+// MustRegisterExtension registers the extensions type with given name
+// and returns a function to get a pointer to the extensions E from a *Limits instance.
+// The name will be used as YAML/JSON key to decode the extensions.
+// This method is not thread safe and should be called only during package initialization.
+func MustRegisterExtension[E any](name string) func(*Limits) *E {
+	for _, e := range registeredExtensions {
+		if e.name == name {
+			panic(fmt.Errorf("extension %s already registered", name))
```

Comment:
nit: also maybe document this panic on duplicate names

Reply:
Documented, thanks.